experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs

```diff
@@ -0,0 +1,109 @@
+
+        public LivePersonAgentBot(ConversationState conversationState)
+        {
+            _conversationState = conversationState;
+        }
+
+        static Dictionary<string, string> Capitals = new Dictionary<string, string>
+        {
+            ["France"] = "Paris",
+            ["Italy"] = "Rome",
+            ["Japan"] = "Tokyo",
+            ["Poland"] = "Warsaw",
+            ["Germany"] = "Hamburg" // not really
+        };
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+            var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+            var userText = turnContext.Activity.Text.ToLower();
+            if (userText.Contains("agent"))
+            {
+                await turnContext.SendActivityAsync("Your request will be escalated to a human agent");
+
+                var transcript = new Transcript(conversationData.ConversationLog.Where(a => a.Type == ActivityTypes.Message).ToList());
+
+                var evnt = EventFactory.CreateHandoffInitiation(turnContext,
+                    new { Skill = "credit-cards" },
+                    transcript);
+
+                await turnContext.SendActivityAsync(evnt);
+            }
+            else
+            {
+                string replyText = $"Sorry, I cannot help you.";
+                if (userText == "hi")
+                {
+                    replyText = "Hello!";
+                }
+                else
+                {
+                    foreach (var country in Capitals.Keys)
+                    {
+                        if (userText.Contains(country.ToLower()))
+                        {
+                            replyText = $"The capital of {country} is {Capitals[country]}";
+                        }
+                    }
+                }
+
+                await turnContext.SendActivityAsync(MessageFactory.Text(replyText, replyText), cancellationToken);
+            }
+        }
+
+        protected override async Task OnEventAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if(turnContext.Activity.Name == HandoffEventNames.HandoffStatus)
+            {
+                var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+                var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+                Activity replyActivity;
+                var state = (turnContext.Activity.Value as JObject)?.Value<string>("state");
+                if (state == HandoffStates.Typing)
+                {
+                    replyActivity = new Activity
```

Comment:
consider pulling through the locale of the original activity.

Reply:
will consider in the future